### PR TITLE
SC-4636

### PR DESCRIPTION
**Pull request by andy8052**
This is an updated circuit breaker logic. 

The code has 4 levels of `breaker`:
0: no breaker
1: no new kick()
2: no new warm() or take()
3: yank() can be called

Any new logic can pass a breaker level (a uint256) and a boolean (higher or lower than a level) to `isStopped` (this probably needs a new name but I really suck at function naming. This allows for a very modular and governance controlled breaker system for the auctions. Maybe it is too open ended?

### CODE REVIEW

File: src/test/clip.t.sol [hunk 1]
```diff
@@ -801,4 +801,80 @@ contract ClipperTest is DSTest {
...
+        hevm.warp(startTime + 3600 seconds);
+        assertTrue(!try_redo(1));
+        hevm.warp(startTime + 3601 seconds);
+        assertTrue( try_redo(1));
+    }
+
+    // does nothing but we can still test the breaker works
+    function test_yank() public {
+        clip.setBreaker(3);
+        clip.yank();
```
Comment (kmbarry1) [inline src/test/clip.t.sol:873]: As I mentioned on your other PR, `yank` shouldn't require the circuit breaker to be turned on. Or, at least there should be a version of it that is just `auth` for the `End` to call. You can, for example, make a wrapper around it and have that require the breaker to be on (so long as you also `rely` the contract on itself). Or just have two functions.

Comment (andy8052) [reply src/test/clip.t.sol:873]: Ah yeah my bad forgot about this

**Review by lucas-manuel — APPROVED** [review]
LGTM

File: src/clip.sol [hunk 2]
```diff
@@ -124,6 +130,11 @@ contract Clipper {
         locked = 0;
     }
 
+    modifier isStopped(uint256 level, bool which) {
```
Comment (kmbarry1) [inline src/clip.sol:133]: I really don't think the second arg is necessary now that we have the levels...we're just passing it as true everywhere anyway.

Comment (andy8052) [reply src/clip.sol:133]: Yeah so with yank it was false. If we ever wanted a yank where we were say it has to be 3 or more then we want it. I was unsure on whether or not I wanted to keep it

Comment (kmbarry1) [reply src/clip.sol:133]: I would keep things simple--there's no need for it right now, it makes the code harder to read/less gas efficient, and we can add it back if we have to. `yank` could always implement some more custom condition without a modifier as a special case.

Comment (andy8052) [reply src/clip.sol:133]: Sounds good will pull it

**Review by kmbarry1 — APPROVED** [review]
LGTM

**Pull request merged**